source/Engine.cpp

```diff
@@ -131,6 +132,18 @@ namespace {
 		}
 	}
 
+	template <class Type>
+	void Prune(list<Type> &objects)
+	{
```

Comment:
```suggestion
	{
		static_assert(derived_from<Body, Type> == true);
```

Reply:
Does it actually need the `== true`?

Reply:
Apparently, `derived_from` is C++20.

Reply:
:disappointed: well then there is probably no way to assert it then

Reply:
There are some older functions that might be good enough. Though I'm not sure if we need this, since none of the other Prune functions have these assertions... I suppose you could make a separate PR cleaning up all these random functions.